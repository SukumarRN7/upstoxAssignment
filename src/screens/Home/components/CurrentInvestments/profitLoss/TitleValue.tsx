import React from 'react';
import {StyleProp, StyleSheet, Text, View, ViewStyle} from 'react-native';

interface ITitleValue {
  title: string;
  value: string;
  style?: StyleProp<ViewStyle>;
}

export const TitleValue: React.FC<ITitleValue> = ({title, value, style}) => {
  return (
    <View style={[styles.container, style]}>
      <Text style={styles.title}>{title}</Text>
      <Text style={styles.value}>{value}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 20,
    flexDirection: 'row',
    justifyContent: 'space-between',
    width: '100%',
  },
  title: {fontWeight: 'bold', fontSize: 16},
  value: {
    fontWeight: '600',
  },
});
