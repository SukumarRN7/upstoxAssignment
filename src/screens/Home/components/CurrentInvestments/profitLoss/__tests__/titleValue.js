import React from 'react';
import renderer from 'react-test-renderer';
import {TitleValue} from '../TitleValue';

describe('TitleValue', () => {
  test('renders correctly', () => {
    const title = 'Title';
    const value = 'Value';
    const component = renderer.create(
      <TitleValue title={title} value={value} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
