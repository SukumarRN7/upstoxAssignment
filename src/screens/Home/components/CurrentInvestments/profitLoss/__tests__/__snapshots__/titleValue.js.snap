// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleValue renders correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 20,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      {
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Title
  </Text>
  <Text
    style={
      {
        "fontWeight": "600",
      }
    }
  >
    Value
  </Text>
</View>
`;
